Test_Cases.py

```diff
@@ -1,169 +1,216 @@
-        self.yahoo_page = Yahoo_Page(self.driver)      
+        options = YahooPage.options
+        s = Service(settings.path_to_webdriver)
+        self.driver = webdriver.Chrome(service=s, options=options)
+        self.yahoo_page = YahooPage(self.driver)
 
     def tearDown(self):
         self.driver.quit()
 
 
+class ErrorMessagePasswords(HomePageSetup):
+
+    def random_password(self, password_len):
+        excluded_password_chars = ['\n', '\t', '\r', '\x0b', '\x0c']
+        return tools.generate_random_string(
+            password_len, excluded_password_chars)
+
+    def setUp(self):
+        super().setUp()
+
+        self.yahoo_page.login()
+        self.yahoo_page.navigate_to_security_tab()
+        self.yahoo_page.click_change_password_link()
+
+    def test_short_password_error_message(self):
+        """
+        Confirm that when the password is 8 characters that the data
+        error status is "WEAK_PASSWORD" Confirm the resulting error text
+        is "- Your password is too easy to guess."
+
+        acceptance criteria
+        --------------------
+        -When an 8 character password is entered, the data error status
+         and the error text exhibit's this is too short of a password.
+
+        Note
+        --------------------
+        -Yahoo security will randomly ask to "prove you are not a robot".
+         Which, at times, can be responsible for the tests in Error Message
+         Passwords class to error. There is not an easy fix for this.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(5))
+
+        self.assertEqual(yahoo_page.short_password_error_message().
+                         get_attribute('data-error'), "WEAK_PASSWORD")
+        self.assertEqual(yahoo_page.short_password_error_message().text,
+                         "- Your password is too easy to guess.")
+
+    def test_moderate_password_error_message(self):
+        """
+        Confirm that when the password 9 or 10 characters in length, the
+        error data status is "ALMOST_THERE". Confirm the resulting error
+        text is "- Almost there".
+
+        acceptance criteria
+        --------------------
+        -When a password that is between 9 and 10 characters is entered,
+         the data error status and the error text exhibit's this is a
+         moderate length for the password.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(9))
+        time.sleep(0.75)  # Please look at the branch Explicit_Wait_ErrorMessagePasswords if you want to see why I used this implicit wait. This keeps my code clean and readable. No E.C.s exist that can wait for this dynamic element correctly for this specific test. So I created my own E.C; which you can see being used in the given branch. This E.C. is also in Tools in this repo.
+
```

Comment:
I'm not familiar with the part of Yahoo you're automating, but I just spent some time playing with my own account and I think I found the text you're looking to validate.

If that's the place, I noticed that when entering a more complex password, there is an element whose class changes dynamically based on the strength of the password.

So, maybe instead of using this static sleep, you should make something like:
`wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, ".error-moderate-level")))`

It may not work, and instead of using `.error-moderate-level` you may need to use a partial attribute (something like `[error-moderate-level]`, but to my point, waiting for this element to be visible using what's really changing in the code means your code will continue only when that event occurs and not when an artificial timer finishes.

Static waits such as `sleep` are a very bad practice because you artificially insert pauses that don't fix the root cause of the problem, so in one hand if the internet is slower than usual your script fails anyway, and in the other, it just keeps waiting even when the event that you're waiting to occur has already happened milliseconds ago. 

When you have multiple tests with these "sleeps" your suite becomes "flaky" and "slow", so you should always try to avoid them by any means.